www/src/components/email-capture-form.js

```diff
@@ -0,0 +1,145 @@
+import React from "react"
+import { rhythm } from "../utils/typography"
+import presets from "../utils/presets"
+import jsonp from "jsonp"
+
+// Mailchimp endpoint
+// From: https://us17.admin.mailchimp.com/lists/integration/embeddedcode?id=XXXXXX
+// Where `XXXXXX` is the MC list ID
+// Note: we change `/post` to `/post-json`
+const MAILCHIMP_URL = `https://gatsbyjs.us17.list-manage.com/subscribe/post-json?u=1dc33f19eb115f7ebe4afe5ee&amp;id=f366064ba7`
+
+class EmailCaptureForm extends React.Component {
+  state = {
+    email: ``,
+  }
```

Comment:
should be set in the constructor eg https://github.com/gatsbyjs/gatsby/blob/master/www/src/components/markdown-page-footer.js#L27-L30

Reply:
Although React docs do say to initialize state within the constructor, I've been reading a lot about how that's not necessary if you don't have props, like here. https://stackoverflow.com/a/37788410 and this post touches on this concept as well: https://babeljs.io/blog/2015/06/07/react-on-es6-plus

But happy to refactor using the constructor.  Just thought I'd share anyway.